common/types.go

```diff
@@ -551,6 +551,27 @@ func (l Location) SubLocation() int {
 	}
 }
 
+func (l Location) SubInSlice(slice Location) Location {
```

Comment:
what does this function do?. A short comment would be helpful.

Reply:
This function returns the location of your subordinate node (if you have a subordinate) within the given slice.

Examples:
* if prime calls `SubInSlice(Location{0,0})` the result will be `Location{0}`, i.e. region-0's location
* if region-0 calls `SubInSlice(Location{0,0})` the result will be `Location{0,0}`, i.e. zone-0-0's location

I'll add a comment to this function

Reply:
understood, thanks. This helps me understand the Use below better.